lib/src/arithmetic/floating_point/floating_point_values/floating_point_8_value.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// floating_point_8_value.dart
+// Implementation of 8-bit Floating-Point value representations.
+//
+// 2024 October 15
+// Authors:
+//  Max Korbel <max.korbel@intel.com>
+//  Desmond A Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:math';
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// The E4M3 representation of a 8-bit floating point value as defined in
+/// [FP8 Formats for Deep Learning](https://arxiv.org/abs/2209.05433).
+class FloatingPoint8E4M3Value extends FloatingPointValue {
+  /// The exponent width
+  static const int exponentWidth = 4;
+
+  /// The mantissa width
+  static const int mantissaWidth = 3;
+
+  @override
+  @protected
+  int get constrainedExponentWidth => exponentWidth;
+
+  @override
+  @protected
+  int get constrainedMantissaWidth => mantissaWidth;
+
+  /// The maximum value representable by the E4M3 format
+  static double get maxValue => 448.toDouble();
+
+  /// The minimum value representable by the E4M3 format
+  static double get minValue => pow(2, -9).toDouble();
+
+  /// Return if the exponent and mantissa widths match E4M3
+  static bool isLegal(int exponentWidth, int mantissaWidth) =>
+      (exponentWidth == 4) & (mantissaWidth == 3);
+
+  /// Constructor for a double precision floating point value
+  FloatingPoint8E4M3Value(
+      {required super.sign, required super.exponent, required super.mantissa});
+
+  /// [FloatingPoint8E4M3Value] constructor from string representation of
+  /// individual bitfields
+  FloatingPoint8E4M3Value.ofBinaryStrings(
+      super.sign, super.exponent, super.mantissa)
+      : super.ofBinaryStrings();
+
+  /// [FloatingPoint8E4M3Value] constructor from spaced string representation of
+  /// individual bitfields
+  FloatingPoint8E4M3Value.ofSpacedBinaryString(super.fp)
+      : super.ofSpacedBinaryString();
+
+  /// [FloatingPoint8E4M3Value] constructor from a single string representing
+  /// space-separated bitfields
+  FloatingPoint8E4M3Value.ofString(String fp, {super.radix})
+      : super.ofString(fp, exponentWidth, mantissaWidth);
+
+  /// [FloatingPoint8E4M3Value] constructor from a set of [BigInt]s of the
+  ///  binary representation
+  FloatingPoint8E4M3Value.ofBigInts(super.exponent, super.mantissa,
+      {super.sign})
+      : super.ofBigInts();
+
+  /// [FloatingPoint8E4M3Value] constructor from a set of [int]s of the binary
+  /// representation
+  FloatingPoint8E4M3Value.ofInts(super.exponent, super.mantissa, {super.sign})
+      : super.ofInts();
+
+  /// Numeric conversion of a [FloatingPoint8E4M3Value] from a host double
+  factory FloatingPoint8E4M3Value.fromDouble(double inDouble) {
+    if ((inDouble > maxValue) | (inDouble < minValue)) {
```

Comment:
can we move this consideration of max/min into the base `fromDouble` in `FloatingPointLogic`?

Reply:
This is specific to E4M3.  it's also wrong, so I have fixed it (it fails for 0.0 and negatives).